src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1577,13 +1579,13 @@ internal virtual async ValueTask<NpgsqlDataReader> ExecuteReader(bool async, Com
                 {
                     foreach (var batchCommand in InternalBatchCommands)
                     {
-                        batchCommand._parameters?.ProcessParameters(dataSource.SerializerOptions, validateValues: true, batchCommand.CommandType);
+                        batchCommand._parameters?.ProcessParameters(reloadableState, validateValues: true, batchCommand.CommandType);
```

Comment:
Not that we really care about multiplexing but this was a bug. Multiple commands forming a batch could have written parameters with different serializer options if they happened to race a reloadtypes.